lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskServiceImpl.java

```diff
@@ -617,6 +619,55 @@ public void forceDeleteTask(String taskId)
     }
   }
 
+  @Override
+  public BulkOperationResults<String, TaskanaException> setOwnerOfTasks(
+      String owner, List<String> argTaskIds) {
+    if (LOGGER.isDebugEnabled()) {
+      LOGGER.debug(
+          "entry to setOwnerOfTasks(owner = {}, tasks = {})",
+          owner,
+          LoggerUtils.listToString(argTaskIds));
+    }
+    try {
+      taskanaEngine.openConnection();
+      BulkOperationResults<String, TaskanaException> bulkLog = new BulkOperationResults<>();
+      if (argTaskIds == null || argTaskIds.isEmpty()) {
+        return bulkLog;
+      }
+      // remove duplicates
+      List<String> taskIds = argTaskIds.stream().sorted().distinct().collect(Collectors.toList());
```

Comment:
why sorted?

Reply:
I had read that distinct() works only correctly on ordered streams. However, this is not true. 
I will remove the sorted() method.
The javadoc says:
Stream<T> distinct()

Returns a stream consisting of the distinct elements (according to Object.equals(Object)) of this stream.
For ordered streams, the selection of distinct elements is stable (for duplicated elements, the element appearing first in the encounter order is preserved.) For unordered streams, no stability guarantees are made.

This is a stateful intermediate operation.

API Note:
    Preserving stability for distinct() in parallel pipelines is relatively expensive (requires that the operation act as a full barrier, with substantial buffering overhead), and stability is often not needed. Using an unordered stream source (such as generate(Supplier)) or removing the ordering constraint with BaseStream.unordered() may result in significantly more efficient execution for distinct() in parallel pipelines, if the semantics of your situation permit. If consistency with encounter order is required, and you are experiencing poor performance or memory utilization with distinct() in parallel pipelines, switching to sequential execution with BaseStream.sequential() may improve performance.
Returns:
    the new stream